brownie/project/main.py

```diff
@@ -296,30 +301,107 @@ def _compile_interfaces(self, compiled_hashes: Dict) -> None:
             self._build._add(abi)
 
     def _load_deployments(self) -> None:
-        if CONFIG.network_type != "live":
+        if CONFIG.network_type != "live" and not CONFIG.settings["dev_deployment_artifacts"]:
             return
-        chainid = CONFIG.active_network["chainid"]
+        chainid = CONFIG.active_network["chainid"] if CONFIG.network_type == "live" else "dev"
         path = self._build_path.joinpath(f"deployments/{chainid}")
         path.mkdir(exist_ok=True)
         deployments = list(path.glob("*.json"))
         deployments.sort(key=lambda k: k.stat().st_mtime)
+        deployment_map: Dict = {}
         for build_json in deployments:
             with build_json.open() as fp:
                 build = json.load(fp)
-            if build["contractName"] not in self._containers:
+
+            contract_name = build["contractName"]
+            if contract_name not in self._containers:
                 build_json.unlink()
                 continue
             if "pcMap" in build:
                 contract = ProjectContract(self, build, build_json.stem)
             else:
                 contract = Contract(  # type: ignore
-                    build["contractName"], build_json.stem, build["abi"]
+                    contract_name, build_json.stem, build["abi"]
                 )
                 contract._project = self
-            container = self._containers[build["contractName"]]
+            container = self._containers[contract_name]
             _add_contract(contract)
             container._contracts.append(contract)
 
+            deployment_map.setdefault(chainid, {}).setdefault(contract_name, []).insert(
+                0, build_json.stem
+            )
+
+        self._save_deployment_map(deployment_map)
+
+    def _clear_dev_deployments(self, height: int = 0) -> None:
+        path = self._build_path.joinpath(f"deployments/dev")
+        if path.exists():
+            deployment_map = self._load_deployment_map()
+            for deployment in list(path.glob("*.json")):
+                if height == 0:
+                    deployment.unlink()
+                else:
+                    with deployment.open("r") as fp:
+                        deployment_artifact = json.load(fp)
+                        block_height = deployment_artifact["deployment"]["blockHeight"]
+                        address = deployment_artifact["deployment"]["address"]
+                        contract_name = deployment_artifact["contractName"]
+                    if block_height > height:
+                        deployment.unlink()
+                        try:
+                            deployment_map["dev"][contract_name].remove(address)
+                        except (KeyError, ValueError):
+                            pass
+            if "dev" in deployment_map and (height == 0 or not deployment_map["dev"]):
+                del deployment_map["dev"]
+                path.rmdir()
```

Comment:
This will raise when the directory is not empty.  Are you absolutely certain it's empty?  Otherwise, maybe better to use `shutil.rmtree`